spring-pulsar/src/main/java/org/springframework/pulsar/core/PulsarTemplate.java

```diff
@@ -92,14 +115,52 @@ public SendMessageBuilder<T> newMessage(T message) {
 		return new SendMessageBuilderImpl<>(this, message);
 	}
 
+	@Override
+	public void setApplicationContext(ApplicationContext applicationContext) {
+		this.applicationContext = applicationContext;
+	}
+
+	@Override
+	public void setBeanName(String beanName) {
+		this.beanName = beanName;
+	}
+
 	/**
-	 * Setter for schema.
+	 * Set the schema to use on this template.
 	 * @param schema provides the {@link Schema} used on this template
 	 */
 	public void setSchema(Schema<T> schema) {
 		this.schema = schema;
 	}
 
+	/**
+	 * Set to true to enable observation via Micrometer.
+	 * @param observationEnabled true to enable.
+	 */
+	public void setObservationEnabled(boolean observationEnabled) {
+		this.observationEnabled = observationEnabled;
+	}
+
+	/**
+	 * Set a custom observation convention.
+	 * @param observationConvention the convention.
+	 */
+	public void setObservationConvention(PulsarTemplateObservationConvention observationConvention) {
+		this.observationConvention = observationConvention;
+	}
+
+	@Override
+	public void afterSingletonsInstantiated() {
+		// TODO is this how we want to do this? What about SBAC?
```

Comment:
Ahh yes.... the green and red of Github diff tool shows me things I forgot to remove before submitting. Will do that shortly.